erigon-lib/kv/membatchwithdb/memory_mutation.go

```diff
@@ -750,3 +751,11 @@ func (m *MemoryMutation) HistoryRange(name kv.Domain, fromTs, toTs int, asc orde
 func (m *MemoryMutation) HistoryStartFrom(name kv.Domain) uint64 {
 	return m.db.(kv.TemporalTx).HistoryStartFrom(name)
 }
+
+func (m *MemoryMutation) EnvInfo() (*mdbxgo.EnvInfo, error) {
```

Comment:
This is leaking abstraction: return underlying data-type. Check - maybe don't need it.  

Also i see existing method:
```
type HasSpaceDirty interface {
	SpaceDirty() (uint64, uint64, error)
}
```